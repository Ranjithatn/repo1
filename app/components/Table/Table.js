import React, { Fragment } from "react";
import Icon from "../Icon/Icon";
import { requestHistorySort } from "../../actions/jobs";
const Table = ({
  className,
  headers,
  requestAscSort,
  data,
  content,
  sortdata,
  requestHistorySort,
  formatMessage,
  disableSort,
}) => {

  // console.log("disableSort",disableSort);

  return (
    <table className={"table" + (className ? " " + className : "")}>
      <thead>
        <tr>
          {headers &&
            headers.length > 0 &&
            headers.map((name, i) => {
              let sorting = true;
              if ( disableSort && disableSort.indexOf(name) != -1 ) {
                sorting = false;
              }

              return (
                <Fragment key={i.toString()}>
                  <th>
                    {name}
                    { sorting &&
                    <Icon
                      icon="angle-up"
                      // onClick={sortdata }
                      onClick={() => {

                        if (sortdata === "job") {

                          requestAscSort({
                            jobsdata: data,
                            rowsdata: name,
                            sorttype: "asc",
                            formatMessage:formatMessage
                          });
                        } else if (sortdata === "history") {
                          requestHistorySort({
                            jobsdata: data,
                            rowsdata: name,
                            sorttype: "asc",
                            formatMessage:formatMessage
                          });
                        }
                      }}
                      // ()=> requestAscSort({
                      //   jobsdata:data,
                      //   rowsdata:name,
                      //   sorttype:"asc"
                      // })
                    />
                    }
                    { sorting &&
                    <Icon
                      icon="angle-down"
                      onClick={() => {
                        if (sortdata === "job") {
                          requestAscSort({
                            jobsdata: data,
                            rowsdata: name,
                            sorttype: "desc",
                            formatMessage:formatMessage
                          });
                        } else if (sortdata === "history") {
                          requestHistorySort({
                            jobsdata: data,
                            rowsdata: name,
                            sorttype: "desc",
                            formatMessage:formatMessage
                          });
                        }
                      }}
                    />
                    }
                  </th>
                </Fragment>
              );
            })}
        </tr>
      </thead>
      <tbody>{content}</tbody>
    </table>
  );
};

export default Table;
