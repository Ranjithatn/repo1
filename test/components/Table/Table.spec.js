import React from "react";
import toJson from "enzyme-to-json";
import Table from "../../../app/components/Table/Table";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

describe('<Table />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Table  />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
