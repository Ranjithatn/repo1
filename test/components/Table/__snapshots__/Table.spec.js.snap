// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render without crashing 1`] = `
<Table>
  <table
    className="table"
  >
    <thead>
      <tr />
    </thead>
    <tbody />
  </table>
</Table>
`;
